source/io/PropertyNode.cpp

```diff
@@ -60,10 +60,45 @@ static bool LookupControlRefForPropertyNode(RefNumVal *refNumPtr, ErrorCluster *
         AddCallChainToSourceIfErrorPresent(errorClusterPtr, propNodeName);
         return false;
     }
-    viName->AppendCStr(vi->VINameCStr());
+    EncodedSubString encodedStr(vi->VIName(), true, false);
+    SubString encodedSubstr = encodedStr.GetSubString();
+    viName->AppendSubString(&encodedSubstr);
     dataItemId->AppendSubString(&dataItemIdSubString);
     return true;
 }
+
+static bool FindVINameAndPropertyPathForValue(StaticTypeAndData *value, ErrorCluster *errorClusterPtr,
+                                            StringRef viName, StringRef path, ConstCStr propNodeName)
+{
+    TypeManagerRef typeManager = value->_paramType->TheTypeManager();
+
+    STACK_VAR(String, pathRef);
+    StringRef symbolPath = pathRef.Value;
+    Boolean foundInVI = false;
+    typeManager->PointerToSymbolPath(value->_paramType, value->_pData, symbolPath, &foundInVI);
+
+    SubString symbolPathSubStr = symbolPath->MakeSubStringAlias();
+    if (symbolPathSubStr.CompareCStr("*pointer-not-found*")) {
```

Comment:
Wondering if instead we should modify `PointerToSymbolPath` to return a boolean instead of always returning `null`